electron/legendary/games.ts

```diff
@@ -561,59 +559,37 @@ class LegendaryGame extends Game {
       rpcClient,
       mangoHudCommand,
       gameModeBin,
-      steamRuntime
-    } = await prepareLaunch(this, gameInfo)
+      steamRuntime,
+      startPlayingDate
+    } = await prepareLaunch(this, mainWindow)
     if (!launchPrepSuccess) {
       appendFileSync(
         this.logFileLocation,
         `Launch aborted: ${launchPrepFailReason}`
       )
+      dialog.showErrorBox(
+        t('box.error.launchAborted', 'Launch aborted'),
+        launchPrepFailReason
+      )
+      await launchCleanup(this, rpcClient, startPlayingDate, mainWindow)
       return false
     }
 
-    const offlineFlag = gameSettings.offlineMode ? '--offline' : ''
+    const offlineFlag = gameSettings.offlineMode ? ['--offline'] : []
```

Comment:
It's a little unnecessary, I agree, but I'd say it's more in-line with all the other args. If you wouldn't have this list here, someone new to the code might want to add a new flag like this: `'--offline --someFlag'` and would then wonder why it isn't working

Reply:
Either way is not understandable for new user. Actually we would need a new function like `setupEnvVars` called `setupLaunchArgs`, which adds the `--offline` to  the launchArgs array.

Reply:
I assume you were planning on sharing this function with GOG's launch method then?

I was specifically avoiding to create a function like that.  
Yes it might reduce the code duplication we have in the two `launch()` functions, but there are certain arguments (like this `--offline` flag for example) that are only present in one function (in this case Legendary), and having *some* launcher args set in a shared function and some set separately is more confusing than just setting them all in `launch()`